crates/orchestrator/src/tests/workers/mod.rs

```diff
@@ -0,0 +1,85 @@
+use crate::config::config_force_init;
+use crate::database::MockDatabase;
+use crate::jobs::types::{ExternalId, JobItem, JobStatus, JobType};
+use crate::queue::job_queue::JobQueueMessage;
+use crate::queue::MockQueueProvider;
+use crate::tests::common::init_config;
+use crate::workers::snos::SnosWorker;
+use crate::workers::Worker;
+use da_client_interface::MockDaClient;
+use httpmock::MockServer;
+use rstest::rstest;
+use serde_json::json;
+use std::collections::HashMap;
+use uuid::Uuid;
+
+#[rstest]
+#[tokio::test]
+async fn test_create_job() {
+    let server = MockServer::start();
+    let da_client = MockDaClient::new();
+    let mut db = MockDatabase::new();
+    let mut queue = MockQueueProvider::new();
+
+    // Mocking db functions
+    db.expect_get_latest_job_by_type().returning(|_| Ok(None)).call(JobType::SnosRun).expect("Failed to call.");
+    // Getting jobs for check expectations
+    for i in 1..6 {
+        db.expect_get_job_by_internal_id_and_type()
+            .returning(|_, _| Ok(None))
+            .call(&i.to_string(), &JobType::SnosRun)
```

Comment:
we aren't supposed to `call` anywhere. instead, we need to tell the mock that this function should be called once (`times(1)`) and it should be called with args xyz (https://docs.rs/mockall/latest/mockall/#matching-arguments)

is this why you were saying that the test doesn't make sense because we're calling everything ourselves?

Reply:
yup

Reply:
actually call behaviour is needed when test is run. 
So what happens is that if I don't add this call then test will throw an error stating that no expectation behaviour found for this particular call. Because this function is called in the code while creating the job to check whether the job already exists, If yes It will throw an error.

Reply:
Hmm, the error sounds like the code didn't call this method but it should've been called. We don't use call for the da test. Can you double or share the exact error message?

Reply:
resolved